src/msgpackdump.d

```diff
@@ -127,229 +132,323 @@ void main(string[] args)
 	ubyte[] bytes = new ubyte[](f.size);
 	f.rawRead(bytes);
 	size_t[] itemCounts;
+	size_t left_padding;
+	string indent = "  ";
+	// Current line
+	enum TextWidth = 4096;
+	char[TextWidth] line;
+	// Current pos in the current line
+	size_t pos;
+
+	void formattedPrintToLine(Char, Args...)(in Char[] fmt, Args args)
+	{
+		import core.exception : RangeError;
+
+		import std.format : sformat;
+		char[] formatted;
+		try
+		{
+			formatted = sformat(line[pos..TextWidth], fmt, args);
+		}
+		catch(RangeError re)
+		{
+			import std.algorithm : copy;
+			import std.format : format;
+			// Buffer is insufficent to output,
+			// allocate appropriate length line on the heap
+			// but copy there only data chunk that is placeable
+			// to the buffer
+			formatted = cast(char[])format(fmt, args)[0..TextWidth - pos];
+			copy(formatted, line[pos..TextWidth]);
+		}
+		pos += formatted.length;
+		assert (pos < TextWidth);
+	}
+
+	void refactorMeformattedPrintToLine(Char, Args...)(in Char[] fmt, Args args)
```

Comment:
Is the "Meformatted" part a typo? Why is "refactor" part of the name?

Reply:
Oh, this is the artifact I missed. I'll fix it.